pkg/cosign/cosign.go

```diff
@@ -137,6 +138,11 @@ func VerifySignature(opts Options) (digest string, err error) {
 		return "", err
 	}
 
+	issuer, _ := extractIssuer(opts.ImageRef, verified, log)
+	if issuer != "" && (issuer != opts.Issuer) {
```

Comment:
We want to support Wildcards in the issuer declaration. So the match needs to be something like `wildcards.match(opts.Issuer, issuer)`.

Reply:
I'm not sure on the practical use case of a wildcard issuer address. Feels dangerous?

Reply:
Its definitely optional. 

To support wildcard subjects would require some changes in Cosign. Will investigate and discuss.

Reply:
We can revisit after more discussions....

Reply:
I checked Cosign and it allows passing in an empty `""` string for the `cosignOpts.CertEmail`:

> CertEmail is the email expected for a certificate to be valid. The empty string means any certificate can be valid.

So, looks like we can set it to empty if the a wildcard is passed, and then perform the check in Kyverno!

@Namanl2001 - can you please try this out?

Reply:
@JimBugwadia when we verify an image we get the output like this [taken from https://www.appvia.io/blog/tutorial-keyless-sign-and-verify-your-container-images]:
```
"Issuer": "https://token.actions.githubusercontent.com",
"Subject": "https://github.com/chrisns/cosign-demo-spotthedifference/.github/workflows/ci.yml@refs/heads/main"
```

so I think `cosignOpts.CertEmail` and `Subject` are two different things and we could check `subject` same as `issuer`

Reply:
@Namanl2001 - see: https://github.com/kyverno/kyverno/blob/main/pkg/cosign/cosign.go#L105.